server/v2/stf/stf.go

```diff
@@ -115,9 +115,17 @@ func (s STF[T]) validateTx(ctx context.Context, state store.WritableState, gasLi
 }
 
 // execTx executes the tx messages on the provided state. If the tx fails then the state is discarded.
-func (s STF[T]) execTx(ctx context.Context, state store.WritableState, gasLimit uint64, tx T) ([]Type, uint64, []event.Event, error) {
+func (s STF[T]) execTx(ctx context.Context, state store.WritableState, gasLimit uint64, tx T) (resp []Type, gasUsed uint64, events []event.Event, err error) {
```

Comment:
nit: For more complex functions like this one, I would personally try to avoid named returns -- they make the code harder to read. Not a blocker though.

Reply:
changed it around, we were also recovering from the error to low in the calls